src/main/java/org/hydev/mcpm/client/uninstall/Uninstaller.java

```diff
@@ -0,0 +1,90 @@
+package org.hydev.mcpm.client.uninstall;
+
+import org.hydev.mcpm.client.database.PluginTracker;
+import org.hydev.mcpm.client.injector.LocalJarBoundary;
+import org.hydev.mcpm.client.injector.PluginNotFoundException;
+import org.hydev.mcpm.client.injector.UnloadBoundary;
+
+import java.io.File;
+
+import static org.hydev.mcpm.client.uninstall.UninstallResult.State.*;
+
+
+/**
+ * Uninstall use case interactor
+ *
+ * @author Anushka (https://github.com/aanushkasharma)
+ */
+public class Uninstaller implements UninstallBoundary {
+    private final PluginTracker tracker;
+    private final UnloadBoundary unloader;
+    private final LocalJarBoundary jarFinder;
+
+    /**
+     * Constructor for uninstaller
+     *
+     * @param tracker Local plugin tracker
+     * @param unloader Unload implementation
+     * @param jarFinder Local jar finder
+     */
+    public Uninstaller(PluginTracker tracker, UnloadBoundary unloader, LocalJarBoundary jarFinder) {
+        this.tracker = tracker;
+        this.unloader = unloader;
+        this.jarFinder = jarFinder;
+    }
+
+    @Override
+    public UninstallResult uninstall(UninstallInput input) {
+        // 1. Unload plugin
+        // This will throw PluginNotFoundException if the plugin isn't loaded. If it is loaded,
+        // this will return the jar of the currently loaded plugin, which is the most precise.
+        // Since people may still uninstall a plugin when it's not loaded, we ignore the not
+        // found error here.
+        File jar = null;
+        if (unloader != null) {
```

Comment:
Might be worth marking `unloader` as `@Nullable`. I had no idea I could pass in null from the Javadoc or anything so far.